pkg/controllers/management/aks/test/buildakscccreateobject.json

```diff
@@ -0,0 +1,69 @@
+{
+  "apiVersion": "aks.cattle.io/v1",
+  "kind": "AKSClusterConfig",
+  "metadata": {
+    "creationTimestamp": null,
+    "name": "c-k4mnv",
+    "ownerReferences": [
+      {
+        "apiVersion": "management.cattle.io/v3",
+        "kind": "Cluster",
+        "name": "c-k4mnv",
+        "uid": "75f1abf6-dab3-40d4-a0a5-bf9df4e42271"
+      }
+    ]
+  },
+  "spec": {
+    "authBaseUrl": null,
+    "authorizedIpRanges": null,
+    "azureCredentialSecret": "cattle-global-data:cc-26b5h",
+    "baseUrl": null,
+    "clusterName": "aks-monday",
+    "dnsPrefix": "aks-dns-99922",
+    "dnsServiceIp": null,
+    "dockerBridgeCidr": null,
+    "httpApplicationRouting": null,
+    "imported": false,
+    "kubernetesVersion": "1.23.8",
+    "linuxAdminUsername": "azureuser",
+    "loadBalancerSku": "Standard",
+    "logAnalyticsWorkspaceGroup": null,
+    "logAnalyticsWorkspaceName": null,
+    "monitoring": null,
+    "networkPlugin": "kubenet",
+    "networkPolicy": null,
+    "nodePools": [
+      {
+        "availabilityZones": [
+          "1",
+          "2",
+          "3"
+        ],
+        "count": 1,
+        "enableAutoScaling": false,
+        "maxPods": 110,
+        "mode": "System",
+        "name": "agentpool",
+        "orchestratorVersion": "1.23.8",
+        "osDiskSizeGB": 128,
+        "osDiskType": "Managed",
+        "osType": "Linux",
+        "vmSize": "Standard_DS2_v2"
+      }
+    ],
+    "podCidr": null,
+    "privateCluster": false,
+    "resourceGroup": "aks-resource-group",
+    "resourceLocation": "eastus",
+    "serviceCidr": null,
+    "subnet": null,
+    "tags": {},
+    "virtualNetwork": null,
+    "virtualNetworkResourceGroup": null
+  },
+  "status": {
+    "failureMessage": "",
+    "phase": "",
+    "rbacEnabled": null
+  }
+}
```

Comment:
Can we remove some of the unneeded values in all of these yaml files? Also I'm assuming these tests don't actually need finalizers or timestamps or valid AKS/EKS/GKE provisioning related data, so we should try to remove as much as we can.